### DISCUSSION TITLE
Is it possible to get the bone matrix of a player?

### DISCUSSION THREAD

**Discussion by willjsaint** [saul/demofile#290] [Questions]
I'm trying to find the position of the head bone of players, googling comes up with "m_dwBoneMatrix" but I wasn't able to find anything of that in the API. Is there any way of getting that info?

      "name": "m_dwBoneMatrix",
      "prop": "m_nForceBone",
      "offset": 28,
      "table": "DT_BaseAnimating"

**Comment by saul** [comment]
You can get the position of the player's eyes by doing:

```ts
s.on("weapon_fire", e => {
  const player = demoFile.entities.getByUserId(e.userid)!;
  const footPosition = player.position;
  const eyePosition = {
    ...footPosition,
    z:
      footPosition.z +
      player.getProp("DT_LocalPlayerExclusive", "m_vecViewOffset[2]")
  };

  console.log(`${player.name} shot from`, eyePosition);
});
```

This offset should always be correct (e.g. it will smoothly change if the player is ducking). Let me know how you get on.

**Reply by willjsaint** [reply]
That unfortunately only gets you the center of the model at the height of the head, not the actual head position. I.E. if you're not at a 90 degree angle it wont be accurate.

Thanks for the help though!

**Reply by saul** [reply]
Out of interest what's the application? Unfortunately the bone transform isn't networked as it's trivial to reconstruct on the client - you could do so in your application if you parsed the model file and played the animations (admittedly a lot of work!)

**Reply by willjsaint** [reply]
I'm only trying to calculate how far off a players crosshair is from the head hitbox. I suppose I could make a lookup table for the x and y offsets depending on the angle of the player and the angle of the victim but that seems like too much work especially considering my skill level with this kinda thing
